packages/core/launch.ts

```diff
@@ -613,7 +613,7 @@ export async function generateArguments(options: LaunchOption) {
   const mc = MinecraftFolder.from(resourcePath)
   const cmd: string[] = []
 
-  const { id = randomUUID().replace(/-/g, ''), name = 'Steve' } = options.gameProfile || {}
+  const { id = randomUUID().replace(/-/g, ''), username = 'Steve' } = options.gameProfile.user || {}
```

Comment:
Here the `gameProfile` type should be 

```ts
gameProfile?: {
    name: string
    id: string
}
```

There should no be `user` inside. I wonder did you pass the correct object here?

Reply:
Actually, gameProfile doesn't returns correct value, it's returns this object:

```
{
  accessToken: '8dfede394e4945ee871824019692b980',
  clientToken: '6b6e3e30c71d4ecb85677462c96a2fc7',
  selectedProfile: { id: '5633c816787e336abee855311338fab3', name: 'fdsfdsf' },
  availableProfiles: [ { id: '5633c816787e336abee855311338fab3', name: 'fdsfdsf' } ],
  user: { id: '5633c816787e336abee855311338fab3', username: 'fdsfdsf' }
}
```

There is a sample log of the Minecraft game.

```
17:00:22.931 (GameService)    › [17:00:22] [main/INFO] [cp.mo.mo.Launcher/MODLAUNCHER]: ModLauncher running: args 
[--username, Steve, --version, 1.16.5-forge-36.2.34, --gameDir, /Users/powerm1nt/.config/FranceRP/game, --assetsDir, /Users/powerm1nt/.config/FranceRP/game/assets, --assetIndex, 1.16, --uuid, 5633c816787e336abee855311338fab3, --accessToken, ❄❄❄❄❄❄❄❄, --userType, Mojang, --versionType, release, --launchTarget, fmlclient, --fml.forgeVersion, 36.2.34, --fml.mcVersion, 1.16.5, --fml.forgeGroup, net.minecraftforge, --fml.mcpVersion, 20210115.111550]
```

With this PR, I fixed this bug, now the log returns the right user:

```
17:00:22.931 (GameService)    › [17:00:22] [main/INFO] [cp.mo.mo.Launcher/MODLAUNCHER]: ModLauncher running: args
[--username, Powerm1nt, --version, 1.16.5-forge-36.2.34, --gameDir, /Users/powerm1nt/.config/FranceRP/game, --assetsDir, /Users/powerm1nt/.config/FranceRP/game/assets, --assetIndex, 1.16, --uuid, 5633c816787e336abee855311338fab3, --accessToken, ❄❄❄❄❄❄❄❄, --userType, Mojang, --versionType, release, --launchTarget, fmlclient, --fml.forgeVersion, 36.2.34, --fml.mcVersion, 1.16.5, --fml.forgeGroup, net.minecraftforge, --fml.mcpVersion, 20210115.111550]
```

You can see, the default username ("Steve") is still used even I set a authentication profile passed on the launch() function.

There is a sample code of a offline authentication system.

```javascript
  async auth (userProfile) {
    log.info(`Authenticating as ${userProfile.username}`)
    this.currentUserProfile = userProfile
    this.authentication = offline(userProfile.username)
    log.info(`Authenticated as ${userProfile.username}`)
    log.debug(this.authentication)
    return userProfile
  }

  async launch () {
    // Launch & Prepare
    return await this.play(
      this.preferredProfile,
      this.authentication,
      this.gamePath,
      this.javaPath
    ).catch(e => log.error(e))
  }
  ```

Reply:
The game profile is the `this.authentication.selectedProfile` which is `{ id: '5633c816787e336abee855311338fab3', name: 'Powerm1nt' }`.

The current offline interface structure design is from the legacy yggdrasil profile API. In yggdrasil API, one user can have multiple gameProfiles (multiple avatar).

So in your case it should be 
```ts
async launch () {
    // Launch & Prepare
    return await this.play(
      this.preferredProfile,
      this.authentication.selectedProfile,
      this.gamePath,
      this.javaPath
    ).catch(e => log.error(e))
  }

```

Please notice this interface is not applicable for new Microsoft authentication.